spyder/config/base.py

```diff
@@ -144,15 +144,15 @@ def get_conf_path(filename=None):
 
     # Create conf_dir
     if not osp.isdir(conf_dir):
-        if running_under_pytest():
+        if running_under_pytest() or TEST:
             os.makedirs(conf_dir)
         else:
             os.mkdir(conf_dir)
     if filename is None:
         return conf_dir
-    else:

```

Comment:
This needs to stay, so please revert this change. It's how we retrieve files placed placed under our config directory, like this `get_conf_path('history.py')`

Reply:
I can, but there is no functional difference to dropping the ``else,`` it works exactly the same (or I would have noticed immediately since this was exactly the functionality I was testing)—did you scroll down a few lines? if filename is not None, it just executes the next statement, which is ``return osp.join(conf_dir, filename)`` exactly as before. I just cleaned it up while I was at it since it was in the same function I made other more substantial changes to.

Reply:
This is a critical part of our settings infrastructure, so I don't want to play with it simply for correctness, so please revert this change.

Reply:
Okay, done as requested. 

In general, I abstain from making these sorts of changes (that aren't purely code style) for the reasons you allude to, namely the small but non-negligible chance of breaking something that's rarely worth the lesser benefit from cleaner code. In this case I ended up doing so since I'd checked it with both my own knowledge and the static analyzer, and the fact that the unit tests and numerous real world-tests exercising that block of code all had no issues, but it obviously wasn't worth doing and I'll be even more conservative in the future.